### PR TITLE
feat(stepfunctions): execution history logging options

### LINKED ISSUE #5754: Enable logging for express stepfunctions
Express stepfunctions work differently than the old ones, the logs are directly written to cloudwatch logs, if enabled. It would be amazing if there was a property on StateMaschineProps to activate logs.

### Use Case

For everyone using express stepfunctions and needing observability 

### Proposed Solution

Add it to the higher level construct

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by workeitel**
## Commit Message
feat(stepfunctions): execution history logging options (#6933)

closes #5754
## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts [hunk 1]
```diff
@@ -24,6 +25,56 @@ export enum StateMachineType {
...
+}
+
+/**
+ * Defines what execution history events are logged and where they are logged.
+ */
+export interface LoggingConfiguration {
+  /**
+   * An array of objects that describes where your execution history events will be logged. Limited to size 1.
+   */
+  readonly destinations: logs.ILogGroup[];
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]: If it's limited to size 1, can we not make this an array.

File: packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts [hunk 2]
```diff
@@ -24,6 +25,56 @@ export enum StateMachineType {
...
 }
 
+/**
+ * Defines which category of execution history events are logged.
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/cloudwatch-log-level.html
+ *
+ * @default ERROR
+ */
+export enum LoggingLevel {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]:
```suggestion
export enum LogLevel {
```

File: packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts [hunk 3]
```diff
@@ -24,6 +25,56 @@ export enum StateMachineType {
...
+    /**
+     * Log fatal errors
+     */
+    FATAL = 'FATAL'
+}
+
+/**
+ * Defines what execution history events are logged and where they are logged.
+ */
+export interface LoggingConfiguration {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]:
```suggestion
export interface LogOptions {
```

File: packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts [hunk 4]
```diff
@@ -60,6 +111,13 @@ export interface StateMachineProps {
      * @default StateMachineType.STANDARD
      */
     readonly stateMachineType?: StateMachineType;
+
+    /**
+     * Defines what execution history events are logged and where they are logged.
+     *
+     * @default No logging
+     */
+    readonly loggingConfiguration?: LoggingConfiguration;
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]:
```suggestion
    readonly logs?: LogOptions;
```

File: packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts [hunk 5]
```diff
@@ -132,11 +190,37 @@ export class StateMachine extends StateMachineBase {
...
+                "logs:GetLogDelivery",
+                "logs:UpdateLogDelivery",
+                "logs:DeleteLogDelivery",
+                "logs:ListLogDeliveries",
+                "logs:PutResourcePolicy",
+                "logs:DescribeResourcePolicies",
+                "logs:DescribeLogGroups"
+              ],
+              resources: ["*"]
+            }));
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]: We try our best to keep least privilege, so avoid `*` resource.

Use `loggroup.grant()` API instead.

File: packages/@aws-cdk/aws-stepfunctions/test/integ.stepfunctions-logging.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import * as logs from '@aws-cdk/aws-logs';
+import * as cdk from '@aws-cdk/core';
+import * as stepfunctions from "../lib";
+
+const app = new cdk.App();
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/test/integ.stepfunctions-logging.ts]: We don't need an integration test for this. Unit tests are sufficient.

File: packages/@aws-cdk/aws-stepfunctions/test/test.state-machine.ts [hunk 7]
```diff
@@ -63,5 +64,66 @@ export = {
         }));
 
         test.done();
-    }
+    },
+
+    'Add log configuration'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/test/test.state-machine.ts]:
```suggestion
    'log configuration'(test: Test) {
```

**Comment by nija-at** [comment]
Thanks for your contribtution!

Once you have addressed these changes and the PR is ready for review, [dismiss my review](https://help.github.com/en/articles/dismissing-a-pull-request-review) and it'll show up in my inbox.

File: packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts (hunk 5 above)
Comment (workeitel) [reply packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]: Those permissions are not bound to resources but more general CloudWatch Logs permissions allowing StepFunctions to write into the account: https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazoncloudwatchlogs.html

There is no resource which can be referenced.

Comment (workeitel) [reply packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts]: As I mentioned the permissions are not bound to a resource but for the whole account. That's also shown in the docs I linked. It does not work with a LogGroup/LogStream restriction.

**Comment by nija-at** [comment]
@workeitel - I'm going to approve this PR again. Please don't click on 'Update branch'. The bot will take care of everything.

**Pull request merged**